js/classic/backend.js

```diff
@@ -0,0 +1,237 @@
+(function(exports) {
+  'use strict';
+
+  exports.classic = exports.classic || {};
+  exports.classic.backend = exports.classic.backend || {};
+
+  var ui = exports.classic.ui;
+  var backend = exports.classic.backend;
+
+  backend.BluetoothManagerTab = function(tabsManager) {
+    this._tabsManager = tabsManager;
+    this._tab = new ui.Tab({}, 'BluetoothManager');
+    this._adapterBlocks = [];
+
+    this._tabsManager.addTab(this._tab);
+
+    var events = ['adapteradded', 'adapterremoved', 'attributechanged'];
+    for (var i = 0; i < events.length; ++i) {
+      navigator.mozBluetooth.addEventListener(
+          events[i], this._updateBlocks.bind(this));
+    }
+  };
+
+  backend.BluetoothManagerTab.prototype = {
+    _updateBlocks: function() {
+      for (var i = 0; i < this._adapterBlocks.length; ++i) {
+        this._adapterBlocks[i].destroy();
+      }
+      this._adapterBocks = [];
+      var adapters = navigator.mozBluetooth.getAdapters();
+      for (i = 0; i < adapters.length; ++i) {
+        this._adapterBlocks.push(new backend.BluetoothManagerTab.AdapterBlock(
+            this._tabsManager, this._tab, adapters[i]));
+      }
+    }
+  };
+
+  backend.BluetoothManagerTab.AdapterBlock = function(
+      tabsManager, tab, adapter) {
+    this._tabsManager = tabsManager;
+    this._tab = tab;
+    this._adapter = adapter;
+    this._block = new ui.PlayButtonBlock(
+        {onPlay: this._onPlayHandler.bind(this)}, '', '', true);
+    this._attributeChangedHandler = this._generateAttributeChangedHandler();
+
+    this._tab.addBlock(this._block, this._tab.numBlocks);
+    this._adapter.addEventListener('attributechanged',
+                                   this._attributeChangedHandler);
+    this._attributeChangedHandler({attrs: ['address']});
```

Comment:
Because at first the default title and descriptions in the new adapter block are both empty strings and we need to trigger that event to initialize them.